### ISSUE TITLE
Log Soap Request for all operation.

### ISSUE THREAD

**Issue by Vijaysinh** [googleads/googleads-php-lib#488]
I am downloading adwords api report using php adwords library.
I wants to keep soap api request log in file.
I am trying bellow code.

use Monolog\Logger;
use Monolog\Handler\StreamHandler;
use Monolog\Formatter\LineFormatter;

$log = new Logger('channel_name');
$stream = new StreamHandler(__DIR__.'/myRequest.log', Logger::INFO);
$stream->setFormatter(new LineFormatter(null, null, false, true));
$log->pushHandler($stream);

$session = (new AdWordsSessionBuilder())
        ->withDeveloperToken(DEVELOPERTOKEN)
        ->withOAuth2Credential($oAuth2Credential)
        ->withClientCustomerId($client_id)
        ->withSoapSettings($SoapSettingsBuilder)
        ->withReportSettings($reportSettings)
        ->withSoapLogger(new Logger('channel_name'))
        ->build();

but its not creating any log files or request.
Could you please help to log my all soap request using php lib without .ini file??
If you provide me any example to log request than it would be great.
I am not using .ini file to log it automatically.

Thanks.

**Comment by fiboknacky** [comment]
At first glance, it looks like you created `$log`, but did not use it with withSoapLogger()?

**Comment by Vijaysinh** [comment]
I tried by adding $log with withSoapLogger() but still Its not working...

$session = (new AdWordsSessionBuilder())
        ->withDeveloperToken(DEVELOPERTOKEN)
        ->withOAuth2Credential($oAuth2Credential)
        ->withClientCustomerId($client_id)
        ->withSoapSettings($SoapSettingsBuilder)
        ->withReportSettings($reportSettings)
        ->withSoapLogger($log)
        ->build();

**Comment by fiboknacky** [comment]
You'd need to check your file permission too.
Instead of creating your own logger from scratch, alternatively, you can use [createLogger()](https://github.com/googleads/googleads-php-lib/blob/0c730de1ed296aefd945ee83eb668cac0a8d71d9/src/Google/AdsApi/Common/AdsLoggerFactory.php?utf8=%E2%9C%93#L41) too.

**Comment by Vijaysinh** [comment]
File permission is fine because I am using same directory for other error log.. Could you please provide any example code to use createLogger() to log adwords request?

**Comment by Vijaysinh** [comment]
Its throwing error that createLogger is non static method... I tried by adding static keyword in front createLogger but no luck, also I changed Logger:INFO to Logger::INFO.

there is no example that how to use logger.

**Comment by fiboknacky** [comment]
Sorry, it should be non-static method. I updated the code snippet.

> I tried by adding static keyword in front createLogger but no luck, also I changed Logger:INFO to Logger::INFO.

What do you mean by no luck? You still face the error or your logger still doesn't work?
If the former, what's the error message?
And what code do you use to test if your logger works?

**Comment by Vijaysinh** [comment]
I mean its not creating any log file... I am using bellow code.
    $adWordsServices = new AdWordsServices();
    $campaignService = $adWordsServices->get($session, CampaignService::class);
    $selector = new Selector();
    $selector->setFields(['Id', 'Name']);
    $selector->setOrdering([new OrderBy('Name', SortOrder::ASCENDING)]);
    $selector->setPaging(new Paging(0,500));
    $totalNumEntries = 0;
    do {
      $page = $campaignService->get($selector);
      $selector->getPaging()->setStartIndex(
          $selector->getPaging()->getStartIndex() + 500);
    } while ($selector->getPaging()->getStartIndex() < $totalNumEntries);
    
$logPath = __DIR__.'/log/myRequest.log';
$adsLoggerFactory = new AdsLoggerFactory();
$logger = $adsLoggerFactory->createLogger('TestChanel',$logPath, Logger::INFO);
pr($logger);

**Comment by Vijaysinh** [comment]
Yes I was passing wrong $logger to my $session... now I am able to log requests.. Thanks @fiboknacky for your support..
